src/python/TaskWorker/Actions/Handler.py

```diff
@@ -40,21 +43,26 @@ def __init__(self, task, procnum, crabserver, config, workFunction, createTempDi
         self.crabserver = crabserver
         self.config = config
         self.workFunction = workFunction
-        self._work = []
-        self._task = task
+        self.work = []
+        self.task = task
         self.taskname = task['tm_taskname']
         self.tempDir = None
         if createTempDir:
             self.tempDir = self.createTempDir()
-            self._task['scratch'] = self.tempDir
+            self.task['scratch'] = self.tempDir
 
         # tm_start_time will be used to set task end time. Make sure
         # it is up to date, in case task site a while in DB e.g. for tape recall.
-        # self._task is passed in input to all actions
-        self._task['tm_start_time'] = int(time.time())
+        # self.task is passed in input to all actions
+        self.task['tm_start_time'] = int(time.time())
+
+        # setup proper credentials for HTCondor
+        tokenDir = getattr(config.TaskWorker, 'SEC_TOKEN_DIRECTORY', None)
+        htcondor.param['SEC_TOKEN_DIRECTORY'] = tokenDir
```

Comment:
I set it here so that it is in a single place rather then inside each Dagman*.py
Indeed it is set it again in the recurring action, I did not change that
https://github.com/belforte/CRABServer/blob/a95b728a9bb207ae0c9e8f7532431f50d81b8eaa/src/python/TaskWorker/Actions/Recurring/RenewRemoteProxies.py#L65-L66

Reply:
I am not sure that I can set the tokendir high enough in the stack that recurring action picks it. IIUC recurring action do not use Handler.py and every action run in a different process. Maybe we could set it in  /etc/condor/config.d and make sure that path is preserved in the container. But let's not overload this PR

Reply:
Agree.  Thanks for pointing out.